File: comicop-backend/src/main/java/com/example/comicop/service/OrderService.java
package com.example.comicop.service;

import com.example.comicop.dto.CreateOrderRequest;
import com.example.comicop.dto.OrderDto;
import java.util.List;

public interface OrderService {

    // Tạo đơn hàng mới từ giỏ hàng hiện tại
    OrderDto createOrder(Long accountId, CreateOrderRequest request);

    // Lấy tất cả đơn hàng của 1 user — mới nhất trước
    List<OrderDto> getOrdersByAccount(Long accountId);

    // Lấy chi tiết 1 đơn hàng
    OrderDto getOrderById(Long orderId, Long accountId);

    // Huỷ đơn hàng — chỉ huỷ được khi status = PENDING
    OrderDto cancelOrder(Long orderId, Long accountId);

    OrderDto confirmPayment(Long orderId, Long accountId);
}

File: comicop-frontend/src/services/orderService.ts
import api from './api'
import type { Order } from '../models/Cart'

// Interface cho request tạo đơn hàng
interface CreateOrderRequest {
  shippingAddress: string
  paymentMethod: string
  note?: string
  voucherCode?: string
}

export const orderService = {
  // Tạo đơn hàng mới — POST /api/orders
  createOrder: (data: CreateOrderRequest) =>
    api.post<Order>('/orders', data),

  // Lấy tất cả đơn hàng của user — GET /api/orders
  getOrders: () => api.get<Order[]>('/orders'),

  // Chi tiết 1 đơn hàng — GET /api/orders/{id}
  getOrder: (orderId: number) =>
    api.get<Order>(`/orders/${orderId}`),

  // Huỷ đơn hàng — POST /api/orders/{id}/cancel
  cancelOrder: (orderId: number) =>
    api.post(`/orders/${orderId}/cancel`),

  // Thêm vào object orderService
  confirmPayment: (orderId: number) =>
    api.post<Order>(`/orders/${orderId}/confirm-payment`),
}